### PR TITLE
Remote Signer implementation

### PR DESCRIPTION
**Pull request by gdbelvin**
This PR implements a authorized account update service.

The service holds a private key that it uses to sign `UpdateEntryRequests`. 
If authorized by each individual account, this service can make updates to their account on their behalf. Eg. This service could update accounts after performing SMS or email based verification. 

The service is structured in the following way: 
1. The service receives a batch account update request 
2. The service verifies that this request is authorized for this app and each user. 
3. The service creates a UpdateEntryRequests
4. The service queues the update requests
5. Return
--- 
The sequencer then:
1. Reads mutations out of the queue
2. Tests to see if they are valid, authorized mutations for any particular account.
3. Applies the mutations to the map and commits by sending the new map head to the log. 
4. Accounts are updated. 

#588

### CODE REVIEW

File: cmd/keytransparency-sequencer/adminapi.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	gwmux := runtime.NewServeMux()
+	if err := ktpb.RegisterKeyTransparencyAdminServiceHandlerFromEndpoint(ctx, gwmux, addr, dopts); err != nil {
+		return nil, err
+	}
+
+	return gwmux, nil
+}
+
+func adminAPI(
+	sqldb *sql.DB,
```
Comment (cesarghali) [inline cmd/keytransparency-sequencer/adminapi.go]: Please put at least one parameter on the first line. It's a bit hard to read this way.

File: core/mutator/entry/mutation.go [hunk 2]
```diff
@@ -28,10 +28,10 @@ import (
 type Mutation struct {
 	userID, appID string
 	index         []byte
-	data, nonce   []byte
+	Data, Nonce   []byte
```
Comment (cesarghali) [inline core/mutator/entry/mutation.go]: I prefer to keep these unexported and add getters.

File: core/mutator/entry/mutation.go [hunk 3]
```diff
@@ -28,10 +28,10 @@ import (
...
 
 	prevEntry *tpb.Entry
-	entry     *tpb.Entry
+	Entry     *tpb.Entry
```
Comment (cesarghali) [inline core/mutator/entry/mutation.go]: Sane here.

File: core/adminservice/server.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	getResp, err := s.tmap.GetLeaves(ctx, &trillian.GetMapLeavesRequest{
+		MapId:    s.mapID,
+		Index:    indexes,
+		Revision: -1,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Update Requests.
```
Comment (cesarghali) [inline core/adminservice/server.go]: Please move what's inside the look to its own unexported function.

File: cmd/keytransparency-sequencer/adminapi.go (hunk 1 above)
Comment (gdbelvin) [reply cmd/keytransparency-sequencer/adminapi.go]: done

**Comment by gdbelvin** [comment]
I've pulled out a bit of the refactor contained this PR into it's own pull request. Please review #817 first.

The rest of this PR has been cleaned up and is ready for a second pass.

File: core/adminservice/server.go (hunk 4 above)
Comment (gdbelvin) [reply core/adminservice/server.go]: done

File: core/mutator/entry/mutation.go (hunk 2 above)
Comment (gdbelvin) [reply core/mutator/entry/mutation.go]: done

File: core/mutator/entry/mutation.go (hunk 3 above)
Comment (gdbelvin) [reply core/mutator/entry/mutation.go]: done

**Pull request closed without merging**